src/classes/sequelize-service.ts

```diff
@@ -97,6 +99,22 @@ export class SequelizeService<W extends {}, R extends W, C extends {} = {}> exte
     }, this.errorFactory);
   }
 
+  public async upsert<KC extends keyof C = keyof {}>(object: W, options: TUpsertOptions<R, C> = {}): Promise<boolean> {
+    return await SequelizeService.try(async () => {
+      return await this.transaction(options, async () => {
+        const session = <ISession<W, R, C>>new CreateSession<W, R, C, KC>([object], options, this);
+
+        await this.executeHook(Hook.WILL_UPSERT, session, this._beforeUpsert.bind(this));
```

Comment:
Erf, I don't like the upsert precisely for that: it's impossible to know which hooks (`WILL_CREATE`/`WILL_UPDATE`) to call :(

I guess there isn't another solution since we're precisely trying not to fetch the instance first.